diff --git a/auditlog/models/rule.py b/auditlog/models/rule.py
index 4dcdb787..86785680 100644
--- a/auditlog/models/rule.py
+++ b/auditlog/models/rule.py
@@ -360,6 +360,8 @@ class AuditlogRule(models.Model):
         log_model = self.env['auditlog.log']
         http_request_model = self.env['auditlog.http.request']
         http_session_model = self.env['auditlog.http.session']
+        if res_model not in self.pool._auditlog_model_cache:
+            return
         for res_id in res_ids:
             model_model = self.env[res_model]
             name = model_model.browse(res_id).name_get()
diff --git a/mass_editing/wizard/mass_editing_wizard.py b/mass_editing/wizard/mass_editing_wizard.py
index 19386b3b..173d7906 100644
--- a/mass_editing/wizard/mass_editing_wizard.py
+++ b/mass_editing/wizard/mass_editing_wizard.py
@@ -66,6 +66,7 @@ class MassEditingWizard(models.TransientModel):
                     })
                     etree.SubElement(xml_group, 'field', {
                         'name': field.name,
+                        'widget': 'many2many_tags',
                         'colspan': '6',
                         'nolabel': '1',
                         'attrs': ("{'invisible': [('selection__" +
